certdb/certdb.go

```diff
@@ -0,0 +1,227 @@
+package certdb
+
+import (
+	"database/sql"
+	"fmt"
+	"time"
+
+	cferr "github.com/cloudflare/cfssl/errors"
+	"github.com/kisielk/sqlstruct"
+)
+
+// CertificateRecord encodes a certificate and its metadata
+// that will be recorded in a database.
+type CertificateRecord struct {
+	Serial    string    `sql:"serial"`
+	CALabel   string    `sql:"ca_label"`
+	Status    string    `sql:"status"`
+	Reason    int       `sql:"reason"`
+	Expiry    time.Time `sql:"expiry"`
+	RevokedAt time.Time `sql:"revoked_at"`
```

Comment:
default zero value, and it is idiomatic to test the zero value by https://golang.org/pkg/time/#Time.IsZero